testsuite/shared/src/main/java/org/jboss/as/test/deployment/DeploymentInfoUtils.java

```diff
@@ -0,0 +1,797 @@
+        check(new InternalParameters(result).setName(name).setSearchType(InternalParameters.SearchType.MISSING).setCtx(ctx));
+    }
+
+    /**
+     * Checking for empty installed deployments.
+     *
+     * @param result Result of called list/info command requested for checking.
+     */
+    public static void checkVoid(DeploymentInfoResult result) {
+        if (result == null) {
+            throw new IllegalArgumentException("Cant check null to void!");
+        }
+        assertThat("Command output contains some deployments! Checking a void FAILED", result.isOutputEmpty(), is(true));
+    }
+
+    /**
+     * In case you need know state of application deployment
+     *
+     * @param result Result of called list/info command requested for checking
+     * @param name   Name of application deployment
+     * @return State of application deployment, if not found return UNKNOWN state
+     * @throws CommandFormatException
+     */
+    public static DeploymentState getStateByOutputMemory(DeploymentInfoResult result, String name) throws CommandFormatException {
+        result.isOutputEmpty();
+
+        for (String row : result.getRows()) {
+            if (row.contains(name)) {
+                final DeploymentState[] statuses = DeploymentState.values();
+
+                for (DeploymentState state : statuses) {
+                    if (row.contains(state.getTitle())) {
+                        log.info("Application deployment are state '" + name + "'->'"
+                                + state.getTitle() + " by command '" + result.getCommand() + " Success");
+                        return state;
+                    }
+                }
+                log.warn("Status of application deployment not found! Do you call info command?\n"
+                        + row);
+                return UNKNOWN;
+            }
+        }
+        throw new CommandFormatException("No result for " + name + " in \n" + result);
+    }
+    // #### END   Method for checking without recalling command for applications deployments state
+
+    // #### BEGIN Internal functionality method
+
+    /**
+     * Calling command in Cli, split output by lines and fill internal memory
+     *
+     * @param command Command for call
+     * @return Readied output, if is output empty return {@value OUTPUT_EMPTY_MARK} mark
+     */
+    private static DeploymentInfoResult callCommand(CLIWrapper cli, String command, String serverGroup) {
+        if (cli == null) {
+            throw new IllegalStateException("Cli is not connected! Call connectCli method first!");
+        }
+
+        cli.sendLine(command);
+        log.info("Called command: '" + command + "'");
+        return new DeploymentInfoResult(command, serverGroup, cli.readOutput());
+    }
+
+    /**
+     * Checking for state or (non)existence of application deployment.
+     *
+     * @param param Containers of parameters, because loot of optional parameters
```

Comment:
because loot of optional parameters ? Not sure it is really required.

Reply:
Do you mean is not required a description or whole class InternalParameters?

Reply:
I mean the documentation: "because loot of optional parameters"